core/src/test/java/io/questdb/cutlass/line/tcp/AlterTableLineTcpReceiverTest.java

```diff
@@ -0,0 +1,445 @@
+    }
+
+    private SqlException sendWithAlterStatement(LineTcpReceiver server, String lineData, String tableName, int wait, boolean noLinger, String alterTableCommand) {
+        sqlException = null;
+        int countDownCount = 1;
+        if ((wait & WAIT_ENGINE_TABLE_RELEASE) != 0 && (wait & WAIT_ALTER_TABLE_RELEASE) != 0) {
+            countDownCount++;
+        }
+        SOCountDownLatch releaseLatch = new SOCountDownLatch(countDownCount);
+        CyclicBarrier startBarrier = new CyclicBarrier(2);
+
+        if (alterTableCommand != null && (wait & WAIT_ALTER_TABLE_RELEASE) != 0) {
+            new Thread(() -> {
+                boolean waited = false;
+                // Wait in parallel thead
+                try {
+                    startBarrier.await();
+                    waited = true;
+                    LOG.info().$("Busy waiting for writer ASYNC event ").$(alterCommandId).$();
+                    sqlException = AlterCommandExecution.waitWriterEvent(
+                            engine,
+                            alterCommandId,
+                            scSequence,
+                            10_000_000,  // 1s
+                            -1
+                    );
+                } catch (Throwable e) {
+                    LOG.error().$(e).$();
+                } finally {
+                    // exit this method if alter executed
+                    releaseLatch.countDown();
+                    if (waited) {
+                        LOG.info().$("Stopped waiting for writer ASYNC event").$();
+                        // If subscribed to global writer event queue, unsubscribe here
+                        AlterCommandExecution.stopEngineAsyncWriterEventWait(engine, scSequence);
+                    }
+                }
+            }).start();
+        }
+
+        if (wait != WAIT_NO_WAIT) {
+            engine.setPoolListener((factoryType, thread, name, event, segment, position) -> {
+                if (Chars.equalsNc(tableName, name)) {
+                    if ((wait & WAIT_ENGINE_TABLE_RELEASE) != 0 || (wait & WAIT_ALTER_TABLE_RELEASE) != 0) {
+                        if (factoryType == PoolListener.SRC_WRITER) {
+                            switch (event) {
+                                case PoolListener.EV_RETURN:
+                                    LOG.info().$("EV_RETURN ").$(name).$();
+                                    releaseLatch.countDown();
+                                    break;
+                                case PoolListener.EV_CREATE:
+                                case PoolListener.EV_GET:
+                                    LOG.info().$("EV_GET ").$(name).$();
+                                    if (alterTableCommand != null) {
+                                        try {
+                                            // Execute ALTER in parallel thread
+                                            alterCommandId = executeAlterSql(alterTableCommand);
```

Comment:
can assert here that command is != -1 rather than asserting inside that `executeAlterSql()` does have alter table command

Reply:
refactored with CompiledQuery api